src/Storefront/Resources/app/storefront/src/plugin-system/plugin.manager.js

```diff
@@ -266,18 +266,62 @@ class PluginManagerSingleton {
         }
 
         if (typeof selector === 'string') {
-            selector = document.querySelectorAll(selector);
+            selector = PluginManagerSingleton._queryElements(selector, pluginClass, options, pluginName);
+            if (!selector) {
+                return;
+            }
         }
 
         return Iterator.iterate(selector, el => {
             PluginManagerSingleton._initializePluginOnElement(el, pluginClass, options, pluginName);
         });
     }
 
+    static _queryElements(selector, pluginClass, options, pluginName) {
+        if (selector.startsWith('#')) {
+            const regexEl = /^#([\w-]+)$/.exec(selector);
```

Comment:
What when the selector is `#foo .bla`?

Reply:
Then iit's using the querySelectorAll

Reply:
https://regex101.com/r/HO4gqC/1 :-)

Reply:
Would it be faster when you look for an empty space, split by empty space and use the first element? So you would not need an regex 🤔

Reply:
I initiale didn't see this, while these are valid selectors:
.foo#ba
 .foo,.bar
.....

Reply:
To restrict the regex to run, I've added a startsWith. I don't see any better, saver and faster way :-(